encode.go

```diff
@@ -287,6 +287,11 @@ func (en *encoder) slice(key uint64, slval reflect.Value) {
 			packed.uvarint(v)
 		}
 
+	case []int:
+		for i := 0; i < sllen; i++ {
+			packed.svarint(int64(slt[i]))
+		}
+
 	case []int32:
 		for i := 0; i < sllen; i++ {
 			packed.svarint(int64(slt[i]))
```

Comment:
> How does the official protobuf do it?

In original protobuf you basically forced to choose: there is no `int` in the .proto language definition: https://developers.google.com/protocol-buffers/docs/proto#scalar (<- see .proto type vs Go type)

The most consistent and reasonable approach would be to simply not allow golang's `int`. But if we want to allow that we should always make sure that on the receiving side there can't be an overflow and else panic.